src/fireedge/src/modules/constants/index.js

```diff
@@ -0,0 +1,243 @@
+
+/**
+ * Server side constants (not all of them are used in client)
+ * Check `window.__PRELOADED_CONFIG__` in src/server/routes/entrypoints/App.js
+ *
+ * @type {object} - Server configuration
+ */
+export const SERVER_CONFIG = (() => {
+  if (isBackend()) return {}
+
+  const config = { ...(window.__PRELOADED_CONFIG__ ?? {}) }
+  delete window.__PRELOADED_CONFIG__
+
+  return config
+})()
+
+export const UNITS = {
+  KB: 'KB',
+  MB: 'MB',
+  GB: 'GB',
+  TB: 'TB',
+  PB: 'PB',
+  EB: 'EB',
+  ZB: 'ZB',
+  YB: 'YB',
+}
+
+// should be equal to the apps in src/server/utils/constants/defaults.js
+export const _APPS = { sunstone: 'sunstone' }
+export const APPS = Object.keys(_APPS)
+
+export const APPS_IN_BETA = []
+export const APPS_WITH_SWITCHER = [_APPS.sunstone]
+
+export const APP_URL = '/fireedge'
+export const WEBSOCKET_URL = `${APP_URL}/websockets`
+export const STATIC_FILES_URL = `${APP_URL}/client/assets`
+export const IMAGES_URL = `${STATIC_FILES_URL}/images`
+export const LOGO_IMAGES_URL = `${IMAGES_URL}/logos`
+export const DEFAULT_IMAGE = `${IMAGES_URL}/default.webp`
+export const IMAGE_FORMATS = ['webp', 'png', 'jpg']
+
+export const FONTS_URL = `${STATIC_FILES_URL}/fonts`
+
+export const SCHEMES = Setting.SCHEMES
+export const DEFAULT_SCHEME = Setting.SCHEMES.SYSTEM
+
+export const CURRENCY = SERVER_CONFIG?.currency ?? 'EUR'
+export const DEFAULT_LANGUAGE = SERVER_CONFIG?.default_lang ?? 'en'
+export const LANGUAGES_URL = `${STATIC_FILES_URL}/languages`
+export const VM_EXTENDED_POOL = !!(SERVER_CONFIG?.use_extended_vmpool ?? true)
+export const LANGUAGES = SERVER_CONFIG.langs ?? {
+  bg_BG: 'Bulgarian (Bulgaria)',
+  bg: 'Bulgarian',
+  ca: 'Catalan',
+  cs_CZ: 'Czech',
+  da: 'Danish',
+  de_CH: 'German (Switzerland)',
+  de: 'German',
+  el_GR: 'Greek (Greece)',
+  en: 'English',
+  es_ES: 'Spanish',
+  et_EE: 'Estonian',
+  fa_IR: 'Persian (Iran)',
+  fa: 'Persian',
+  fr_CA: 'French (Canada)',
+  fr_FR: 'French',
+  hu_HU: 'Hungary',
+  it_IT: 'Italian',
+  ja: 'Japanese',
+  ka: 'Georgian',
+  lt_LT: 'Lithuanian',
+  nl_NL: 'Dutch',
+  pl: 'Polish',
+  pt_PT: 'Portuguese',
+  ro_RO: 'Romanian',
+  ru_RU: 'Russian',
+  ru: 'Russian',
+  si: 'Sinhala',
+  sk_SK: 'Slavak',
+  sr_RS: 'Serbian',
+  sv: 'Swedish',
+  th_TH: 'Thai (Thailand)',
+  th: 'Thai',
+  tr_TR: 'Turkish (Turkey)',
+  tr: 'Turkish',
+  uk_UA: 'Ukrainian (Ukraine)',
+  uk: 'Ukrainian',
+  vi: 'Vietnamese',
+  zh_CN: 'Chinese (China)',
+  zh_TW: 'Chinese (Taiwan)',
+}
+
+export const ONEADMIN_ID = '0'
+export const SERVERADMIN_ID = '1'
+export const ONEADMIN_GROUP_ID = '0'
+
+export const FILTER_POOL = {
+  PRIMARY_GROUP_RESOURCES: '-4',
+  USER_RESOURCES: '-3',
+  ALL_RESOURCES: '-2',
+  USER_GROUPS_RESOURCES: '-1',
+}
+
+/** @enum {string} Input types */
+export const INPUT_TYPES = {
+  AUTOCOMPLETE: 'autocomplete',
+  CHECKBOX: 'checkbox',
+  SWITCH: 'switch',
+  FILE: 'file',
+  TIME: 'time',
+  HIDDEN: 'hidden',
+  PASSWORD: 'password',
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential hardcoded credential detected. This code may contain sensitive data such as passwords or API keys embedded directly in the source.  Hardcoded credentials can be extracted and misused, leading to unauthorized access to systems or data breaches.  To remediate this, store secrets in environment variables or use a secrets management tool like AWS Secrets Manager, Azure Key Vault, or HashiCorp Vault.  Avoid committing credentials to version control. For best practices, refer to - https://owasp.org/www-community/vulnerabilities/Use_of_hard-coded_password